### ISSUE TITLE
Sporadic release failures

### ISSUE THREAD

**Issue by sambooo** [weaveworks/flux#625]
Setup:
1. `minikube start --memory=4096`
2. Clone the [example repo](https://github.com/weaveworks/flux-example)
3. Create the example service: `kubectl apply -f helloworld-deploy.yaml -f helloworld-svc.yaml`
4. Create a Weave Cloud instance and download the manifest: `curl --location "https://cloud.weave.works/k8s.yaml?service-token=[redacted]&k8s-version=$(kubectl version | base64 | tr -d '\n')" > cloud.yml`
5. (Edit and) apply: `kubectl apply -n kube-system -f cloud.yml`
6. Configure flux on Weave Cloud

Observations:
- Attempting to upgrade `helloworld:master-a000002` to `a000001` with Flux UI and `fluxd:0.3.0`
  - Failure in UI:
    ```
    Release job submitted, ID 1c4c60e5-49ff-1612-1649-737260155267
    Status: Applying changes.
    
    1) Queued.
    2) Calculating updates for release.
    3) Cloning git repository.
    4) Finding defined services.
    5) Found service default/helloworld
    6) Found 1 services.
    7) Looking up images.
    8) Will update default/helloworld container helloworld: quay.io/weaveworks/helloworld:master-a000002 -> master-a000001
    9) Pushing changes.
    10) Applying changes.
    Failure
    ```
  - Successfully committed and updated pods
  - Subsequent releases succeed or fail seemingly randomly

**Comment by squaremo** [comment]
I ran through this, and got the result as described, with one important caveat: the first time I ran a release, it pushed a commit but failed to apply the change to the cluster (because of the mismatched kubectl issue of fluxd 0.2.0 vs Kubernetes 1.6). Thus, when I updated fluxd to 0.3.0, the cluster did not reflect what was in the git repo, and releases failed with "no changes" until I manually applied the config from the git repo.

From that point, releases mostly worked. Occasionally, I could catch it out and get a "no changes" failure, during the dry-run phase. This may be because the display, which checks the state of the cluster, may not always be in exact sync with the git repo (i.e., things might not have been successfully applied yet). It's safe to try again, and perhaps we should be clearer about that, and about when we can see there's a difference between what's running and what's configured.
